src/main/java/com/scaleunlimited/flinkcrawler/urls/SimpleUrlLengthener.java

```diff
@@ -49,14 +56,22 @@ public SimpleUrlLengthener(BaseHttpFetcherBuilder fetcherBuilder) {
     public void open() throws Exception {
         _fetcher = _fetcherBuilder.build();
         _urlShorteners = loadUrlShorteners();
+        _lengthenedUrlMap = new LinkedHashMap<RawUrl, RawUrl>(  (LRU_CACHE_CAPACITY * 4/3),
+                                                                LRU_CACHE_LOAD_FACTOR, 
+                                                                true); // LRU
```

Comment:
Hmm - don't you have to override `removeEldestEntry(Map.Entry)` to trigger the expunging of an entry when the cache size is > `LRU_CACHE_CAPACITY`? See https://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html#removeEldestEntry(java.util.Map.Entry)